components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/APIResourceManager.java

```diff
@@ -156,4 +156,13 @@ void putScopes(String apiResourceId, List<Scope> currentScopes, List<Scope> scop
      * @throws APIResourceMgtException If an error occurs while retrieving scope.
      */
     Scope getScopeByName(String scopeName, String tenantDomain) throws APIResourceMgtException;
+
+    /**
+     * Get scope metadata by scope names and tenant domain.
+     *
+     * @param scopeNames   List of scope names.
+     * @param tenantDomain Tenant domain.
+     */
+    List<APIResource> getScopeMetadata(List<String> scopeNames, String tenantDomain)
```

Comment:
if this interface is not yet included in major release, adding new services without `default` modifier is fine, assuming this interface doesn't have any custom implementations. otherwise make this default

Reply:
Since this not yet included in a major release keeping this way.